components/mailgun/mailgun.app.js

```diff
@@ -0,0 +1,192 @@
+const axios = require("axios");
+const get = require("lodash/get");
+const moment = require("moment");
+const retry = require("async-retry");
+
+module.exports = {
+  type: "app",
+  app: "mailgun",
+  propDefinitions: {
+    domain: {
+      type: "string",
+      label: "domain",
+      description: "Domain used to send and receive email.",
+      async options(context) {
+        const q = context.query;
+        const options = [];
+        const results = await this.getAllDomains();
+        for (const domains of results) {
+          options.push(domains.name);
+        }
+        return options;
+      },
+    },
+    baseRegion: {
+      type: "string",
+      label: "Base region",
+      description:
+        'The region that Mailgun will use to send and receive email, which determines the URL to use when requesting Mailgun API. This must be the same region where "domain" was created.',
+      default: "US",
+      options: ["US", "EU"],
+    },
+    timer: {
+      label: "Polling schedule",
+      description: "Pipedream polls Reddit for events on this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15, // by default, run every 15 minutes.
+      },
+    },
+  },
+  methods: {
```

Comment:
For methods in the app file, we're trying to document these well for future devs. IMO there's no need to document private methods like `_apiKey()`, but it would be useful for others. Could you take a pass at adding [JS Docs](https://github.com/PipedreamHQ/pipedream/blob/master/COMPONENT-GUIDELINES.md#methods) to the core, public methods?

Reply:
Interesting. I have created documentation for public methods under the mailgun app file.